### PR TITLE
Added fallthrough check in matching

### PR DESCRIPTION
**Pull request by Mixaster995**
Signed-off-by: Mikhail Avramenko <avramenkomihail15@gmail.com>

## Description

Fix for match selector which enables continuation in case of absent nse candidates

## Issue link
part of https://github.com/networkservicemesh/deployments-k8s/issues/2007
blocked by https://github.com/networkservicemesh/api/pull/106

## How Has This Been Tested?

- [x] Added unit testing to cover
- [x] Tested manually
- [ ] Tested by integration testing
- [ ] Have not tested

## Types of changes

- [x] Bug fix
- [ ] New functionallity
- [ ] Documentation
- [ ] Refactoring
- [ ] CI

### CODE REVIEW

**Comment by denis-tingaikin** [comment]
Please add also unit test to cover the use-case with `fallthrough`

File: pkg/networkservice/chains/nsmgr/suite_test.go [hunk 1]
```diff
@@ -603,7 +668,7 @@ const (
 	labelB = "label_b"
 )
 
-func linearNS(count int) *registry.NetworkService {
+func linearNS(count int, twoEmpty, withPassthrough bool) *registry.NetworkService {
```
Comment (denis-tingaikin) [inline pkg/networkservice/chains/nsmgr/suite_test.go:671]: This looks complicated. Please use this inline for the new test.

Comment (Mixaster995) [reply pkg/networkservice/chains/nsmgr/suite_test.go]: done

File: pkg/networkservice/chains/nsmgr/suite_test.go [hunk 2]
```diff
@@ -513,6 +513,83 @@ func (s *nsmgrSuite) Test_PassThroughLocalUsecase() {
...
+	// Path length to first endpoint is 5
+	// Path length from NSE client to other local endpoint is 5
+	require.Equal(t, 5*(nsesCount-2)+5, len(conn.Path.PathSegments))
+	for i := 1; i < len(nseRegs); i++ {
+		require.Contains(t, nseRegs[i].Name, conn.Path.PathSegments[i*5-1].Name)
+	}
+
+	// Refresh
+	conn, err = nsc.Request(ctx, request)
+	require.NoError(t, err)
```
Comment (denis-tingaikin) [inline pkg/networkservice/chains/nsmgr/suite_test.go:576]:
```suggestion
	require.NoError(t, err)
	require.Equal(t, 5*(nsesCount-2)+5, len(conn.Path.PathSegments))
```

Comment (Mixaster995) [reply pkg/networkservice/chains/nsmgr/suite_test.go:576]: ok, done

**Review by denis-tingaikin — APPROVED** [review]
LGTM

**Pull request merged**